app/snapshot.py

```diff
@@ -77,30 +83,39 @@ def build_model(region_name: str, resource_dict: dict):
         snapshot_type = resource_dict['StorageTier']
         resource_id_tag = 'SnapshotId'
         resource_type_tag = 'StorageTier'
-        monthly_price = resource.estimate_monthly_snapshot_price(snapshot_type, size)
+        is_aws_backup_snapshot = False
+        is_ami_snapshot = False
+        if "AWS Backup service" in resource_dict['Description']:
+            is_aws_backup_snapshot = True
+        elif "Copied for DestinationAmi" in resource_dict['Description']:
```

Comment:
This might not be the most rebust method for Amis in particular because if someone deletes an AMI manually and doesn't delete the snapshot, then nagbot won't ever clean it up.

Reply:
Implemented logic for the following:
If snapshot is an ami snapshot, we use a regex to match the first occurrence of an ami id since the first occurrence refers to the 'Destination AMI' (what we want) instead of the 'Source AMI'.
That ami id is checked against all the orgs amis to see if it is registered - if it is not registered the snapshot is not considered an ami snapshot and it will be treated like the rest of its bretheren able to be deleted. (After running this looks like we have a good amount of ami snapshots whose amis have been deregistered so we should save a nice chunk of change by deleting them!)
One note - I used a global variable for the orgs amis so that the API call would only be made rather than once for every ami snapshot since the call is costly.